src/Tools/Source/RunTests/ProcessRunner.cs

```diff
@@ -109,15 +109,31 @@ public static void OpenFile(string file)
                 // We must call WaitForExit to make sure we've received all OutputDataReceived/ErrorDataReceived calls
                 // or else we'll be returning a list we're still modifying. For paranoia, we'll start a task here rather
                 // than enter right back into the Process type and start a wait which isn't guaranteed to be safe.
-                Task.Run(() =>
+                Task.Run(async () =>
                 {
-                    process.WaitForExit();
+                    int exitCode;
+                    try
+                    {
+                        exitCode = await GetExitCodeAsync(process);
+                    }
+                    catch (Exception ex)
+                    {
+                        tcs.TrySetException(ex);
+                        throw;
+                    }
+
                     var result = new ProcessResult(
                         process,
-                        process.ExitCode,
+                        exitCode,
                         new ReadOnlyCollection<string>(outputLines),
                         new ReadOnlyCollection<string>(errorLines));
                     tcs.TrySetResult(result);
+
+                    static async ValueTask<int> GetExitCodeAsync(Process process)
+                    {
+                        await process.WaitForExitAsync();
```

Comment:
📝 This pull request fixes the case where `WaitForExit` _throws_, but does not do anything about a case where it never returns at all. Currently it appears more likely that problems are caused by throwing.